banzai/qc/pattern_noise.py

```diff
@@ -21,49 +22,197 @@ def group_by_keywords(self):
 
     def do_stage(self, images):
         for image in images:
-            # If the data is a cube, then run on each extension individually
             logging_tags = logs.image_config_to_tags(image, self.group_by_keywords)
             logs.add_tag(logging_tags, 'filename', os.path.basename(image.filename))
             logs.add_tag(logging_tags, 'snr_threshold', self.SNR_THRESHOLD)
-            logs.add_tag(logging_tags, 'freq_bins_threshold', self.FREQ_BINS_THRESHOLD)
+            logs.add_tag(logging_tags, 'std_threshold', self.STD_THRESHOLD)
+
+            # If the data is a cube, then run on each extension individually
             if image.data_is_3d():
-                pattern_noise_is_bad = any([check_for_pattern_noise(data, self.SNR_THRESHOLD, self.FREQ_BINS_THRESHOLD)
-                                            for data in image.data])
+                pattern_noise_is_bad = any([self.check_for_pattern_noise(data) for data in image.data])
```

Comment:
If this runs after the stitching, will it ever be a cube?

Reply:
Nope, good point, I'll take this out.